### PR TITLE
particle designer2.0 support

### PR DESCRIPTION
**Pull request by yuye-liu**
particle designer2.0 support

### CODE REVIEW

**Comment by yuye-liu** [comment]
particle designer2.0 support perfect

File: cocos2dx/particle_nodes/CCParticleSystem.cpp [hunk 1]
```diff
@@ -205,6 +207,11 @@ bool ParticleSystem::initWithDictionary(Dictionary *dictionary, const char *dirn
         // self, not super
         if(this->initWithTotalParticles(maxParticles))
         {
+			// Emitter name in particle designer 2.0
+			const String * configNameConstStr = dictionary->valueForKey("configName");
+            m_configName = configNameConstStr->getCString();
+			//sprintf(m_configName, "%s", dictionary->valueForKey("configName").c_str());
```
Comment (minggo) [inline cocos2dx/particle_nodes/CCParticleSystem.cpp]: Error indent here and following modifications.

File: cocos2dx/particle_nodes/CCParticleSystem.cpp [hunk 2]
```diff
@@ -116,6 +117,7 @@ of this software and associated documentation files (the "Software"), to deal
 , _blendFunc(BlendFunc::ALPHA_PREMULTIPLIED)
 , _opacityModifyRGB(false)
 , _positionType(PositionType::FREE)
+, m_yCoordFlipped(0)
```
Comment (minggo) [inline cocos2dx/particle_nodes/CCParticleSystem.cpp]: `m_yCoordFlipped` -> `_yCoordFlipped`?

File: cocos2dx/particle_nodes/CCParticleSystem.cpp [hunk 3]
```diff
@@ -81,7 +81,8 @@ of this software and associated documentation files (the "Software"), to deal
 //
 
 ParticleSystem::ParticleSystem()
-: _isBlendAdditive(false)
+: m_configName("")
```
Comment (minggo) [inline cocos2dx/particle_nodes/CCParticleSystem.cpp]: `m_configName` -> `_configName`

**Comment by minggo** [comment]
And could you add test case for particle designer 2?

File: cocos2dx/particle_nodes/CCParticleSystem.h [hunk 4]
```diff
@@ -540,6 +543,9 @@ class CC_DLL ParticleSystem : public Node, public TextureProtocol
     BlendFunc _blendFunc;
     /** does the alpha value modify color */
     bool _opacityModifyRGB;
+	/** does FlippedY variance of each particle */
```
Comment (minggo) [inline cocos2dx/particle_nodes/CCParticleSystem.h:546]: Error indent.

File: cocos2dx/particle_nodes/CCParticleSystem.h [hunk 5]
```diff
@@ -456,6 +456,9 @@ class CC_DLL ParticleSystem : public Node, public TextureProtocol
     //! Array of particles
     tParticle *_particles;
 
+	//Emitter name
+	std::string _configName;
```
Comment (minggo) [inline cocos2dx/particle_nodes/CCParticleSystem.h:460]: Error indent.

**Comment by minggo** [comment]
@yuye-liu 
Could you please sent a pull request for mater branch too?

**Comment by yuye-liu** [comment]
I'm cloning the master branch. if downloaded ,i'' push it as soon as possible

------------------ 原始邮件 ------------------
发件人: "minggo";notifications@github.com;
发送时间: 2013年9月24日(星期二) 下午3:41
收件人: "cocos2d/cocos2d-x"cocos2d-x@noreply.github.com; 
抄送: "yuye.liu"yuye.liu@chukong-inc.com; 
主题: Re: [cocos2d-x] particle designer2.0 support (#3689)

@yuye-liu 
 Could you please sent a pull request for mater branch too?

**Pull request merged**